### PR TITLE
Ria 3305 hearing bundle ready

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3305

### Change description ###
RIA-3305  hearing  bundle ready

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandler.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            = listingNotificationHelper.getHearingBundleReadyInstructMessage(
+                asylumCase, notificationsHelper.getConsumerReference(caseId),
+                notificationsHelper.getMessageHeader(), homeOfficeReferenceNumber);
+
+        log.info("Finished constructing {} notification request for caseId: {}, HomeOffice reference: {}",
+            HEARING_BUNDLE_READY.toString(), caseId, homeOfficeReferenceNumber);
+
+        final String notificationStatus = homeOfficeInstructService.sendNotification(hearingInstructMessage);
+
+        log.info("SENT: {} notification for caseId: {}, HomeOffice reference: {}, status: {}",
```
Comment (jkumar-moj) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandler.java:76]: Could you please add the event as well to the logs here for easy traceability?

File: src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/ListingNotificationHelperTest.java [hunk 2]
```diff
@@ -139,6 +139,23 @@ void shouldGetAdjournHearingInstructMessage() {
             hearingInstructMessage.getNote());
     }
 
+    @Test
+    @MockitoSettings(strictness = Strictness.WARN)
+    void shouldGetHearingBundleReadyInstructMessage() {
+
+        Hearing.HearingBuilder hearingData = Hearing.HearingBuilder.hearing()
+            .withHearingType("ORAL")
+            .withHmctsHearingRef("ariaListingReference");
```
Comment (jkumar-moj) [inline src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/ListingNotificationHelperTest.java:148]: I don't see this statement being used anywhere??

File: src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        verify(asylumCase, times(1)).write(HOME_OFFICE_HEARING_BUNDLE_READY_INSTRUCT_STATUS, "OK");
+        verify(homeOfficeInstructService).sendNotification(hearingInstructMessageArgumentCaptor.capture());
+
+        final HearingInstructMessage instructMessage = hearingInstructMessageArgumentCaptor.getValue();
+        assertNotificationInstructMessage(instructMessage);
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Event.class, names = {"GENERATE_HEARING_BUNDLE", "CUSTOMISE_HEARING_BUNDLE"})
+    @MockitoSettings(strictness = Strictness.WARN)
```
Comment (jkumar-moj) [inline src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandlerTest.java:106]: Have you verified if it is really needed here? I remeber it was used initially to overcome an issue. I do not want to just have it to replicate the other tests. Could you please verify it once?

**Review by jkumar-moj — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandler.java (hunk 1 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandler.java]: Added event in this log as well

File: src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/ListingNotificationHelperTest.java (hunk 2 above)
Comment (Lee365) [reply src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/ListingNotificationHelperTest.java]: Removed unused statement.

File: src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandlerTest.java (hunk 3 above)
Comment (Lee365) [reply src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandlerTest.java]: Removed settings and corrected

**Review by jkumar-moj — APPROVED** [review]
👍

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Comment by jkumar-moj** [comment]
Reopening as it was closed by stale bot

**Pull request merged**